pkg/controllers/v1alpha1/dataload/dataload_controller.go

```diff
@@ -91,6 +91,7 @@ func (r *DataLoadReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 	// 2. Reconcile deletion of the object if necessary
 	if utils.HasDeletionTimestamp(dataload.ObjectMeta) {
+		r.RemoveEngine(ctx)
 		return r.ReconcileDataLoadDeletion(ctx, targetDataload)
```

Comment:
I suggest to move `r.RemoveEngine` into `r.ReconcileDataLoadDeletion`. Thanks.

Reply:
thx, done